html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -163,22 +180,194 @@ public function home()
 	{
 		// 自分が落札したBidinfoをページネーションで取得
 		$bidinfo = $this->paginate('Bidinfo', [
-			'conditions'=>['Bidinfo.user_id'=>$this->Auth->user('id')], 
+			'conditions' => ['Bidinfo.user_id' => $this->Auth->user('id')],
 			'contain' => ['Users', 'Biditems'],
-			'order'=>['created'=>'desc'],
-			'limit' => 10])->toArray();
-		$this->set(compact('bidinfo'));
+			'order' => ['created' => 'desc'],
+			'limit' => 10
+		])->toArray();
+		$ratings = $this->Ratings->find()->select('biditem_id')->where(['reviewer_id' => $this->Auth->user('id')])->toArray();
+		if (!empty($ratings)) {
+			foreach ($ratings as $rate) {
+				$endOfTransaction[] = $rate->biditem_id;
+			}
+		} else {
+			$endOfTransaction = array(0);
+		}
+		$this->set(compact('bidinfo', 'endOfTransaction'));
 	}
 
 	// 出品情報の表示
 	public function home2()
 	{
 		// 自分が出品したBiditemをページネーションで取得
 		$biditems = $this->paginate('Biditems', [
-			'conditions'=>['Biditems.user_id'=>$this->Auth->user('id')], 
+			'conditions' => ['Biditems.user_id' => $this->Auth->user('id')],
 			'contain' => ['Users', 'Bidinfo'],
-			'order'=>['created'=>'desc'],
-			'limit' => 10])->toArray();
-		$this->set(compact('biditems'));
+			'order' => ['created' => 'desc'],
+			'limit' => 10
+		])->toArray();
+		$ratings = $this->Ratings->find()->select('biditem_id')->where(['reviewer_id' => $this->Auth->user('id')])->toArray();
+		if (!empty($ratings)) {
+			foreach ($ratings as $rate) {
+				$endOfTransaction[] = $rate->biditem_id;
+			}
+		} else {
+			$endOfTransaction = array(0);
+		}
+		$this->set(compact('biditems', 'endOfTransaction'));
+	}
+
+	// 取引終了後のページ
+	public function interact($id = null)
+	{
+		// bidinfoテーブルに商品idのデータが入っているか検索
+		$bidinfo = $this->Bidinfo->find('all')->where(['biditem_id' => $id])->first();
+
+		// ログインしているユーザーを変数に挿入
+		$loginUserId = $this->Auth->user('id');
+		// 出品者と落札者のuser_idを検索
+		$seller = $this->Biditems->find()->select(['user_id'])->where(['id' => $id])->first();
+		$buyer = $this->Bidinfo->find()->select(['user_id'])->where(['biditem_id' => $id])->first();
+
+		// bidinfoテーブルに商品idのデータが入っていない時
+		if (is_null($bidinfo)) {
+			$this->Flash->success(__('権限がありません'));
+			// トップページ（index）に移動
+			return $this->redirect(['action' => 'index']);
+		}
+		// ログインユーザーが出品者・落札者であるか
+		switch (true) {
+				// 出品者
+			case ($loginUserId === $seller['user_id']):
```

Comment:
```suggestion
			case ($loginUserId === $seller['user_id']): // 出品者
```
こうですね。もしくはこう
```php
			case ($loginUserId === $seller['user_id']):
			// 出品者
```
他のファイルのswitch文も見直し